dask_kubernetes/core.py

```diff
@@ -131,18 +131,26 @@ class Scheduler(Pod):
         The scheduler task will exit after this amount of time
         if there are no requests from the client. Default is to
         never timeout.
+    service_wait_timeout: int (optional)
+        Timeout, in seconds, to wait for the remote scheduler service to be ready.
+        Defaults to 30 seconds.
     """
 
-    def __init__(self, idle_timeout: str, **kwargs):
+    def __init__(self, idle_timeout: str, service_wait_timeout_s: int = None, **kwargs):
         super().__init__(**kwargs)
         self.service = None
-        self._service_wait_timeout_s = None
         self._idle_timeout = idle_timeout
+        self._service_wait_timeout_s = service_wait_timeout_s
 
         self.pod_template.metadata.labels["dask.org/component"] = "scheduler"
         cli_args = ["dask-scheduler"]
         if self._idle_timeout is not None:
-            cli_args += ["--idle-timeout", self.__idle_timeout]
+            cli_args += ["--idle-timeout", self._idle_timeout]
+
+        if self._service_wait_timeout_s is None:
+            self._service_wait_timeout_s = (
+                dask.config.get("kubernetes.scheduler-service-wait-timeout", 30) or 30
+            )
```

Comment:
I would prefer this to be consistent with other options and also moved to the `_start` method.

I would expect to see something there like:

```python
self._scheduler_service_wait_timeout = self._scheduler_service_wait_timeout or dask.config.get("kubernetes.scheduler-service-wait-timeout")
```

We also do not set our default values in code as the Dask config library will draw default values from the [example config file](https://github.com/dask/dask-kubernetes/blob/master/dask_kubernetes/kubernetes.yaml). So if you could add the default of `30` there I would appreciate it.

Reply:
I think I already set the default value in the example config file. I'll move the parsing to `KubeCluster._start` though.

If someone sets the `scheduler-service-wait-timeout` to null in the config, it will crash as it is though.

Reply:
Thanks for making the change.

I feel like if someone sets the timeout to `null` then it should wait forever. Perhaps we should implement this and then document the fact that it's probably not a great idea to do that.

Reply:
Sure. Where would that go in terms of documentation ? In the `Scheduler` docstring or somewhere else ?

Reply:
I would put it in the `KubeCluster` docstring for now with the new kwarg.

Reply:
Hum okay but you can't really set it to None from the KubeCluster constructor since it already defaults to None. This "option" will only be available when using the config file.

Reply:
Maybe we can say that the "no-timeout" value is `-1` and not `None` this way we can pass it also in the `KubecCluster` constructor.

Reply:
I think `0` is a common value for disabling timeouts like this.

Reply:
Ok I'll go with `0` then !